packages/lib/src/components/UIElement.tsx

```diff
@@ -77,7 +77,7 @@ export class UIElement<P extends UIElementProps = any> extends BaseElement<P> im
             component = this.constructor['type'] === 'scheme' || this.constructor['type'] === 'bcmc' ? this.constructor['type'] : this.props.type;
         }
 
-        this.props.modules?.analytics.sendAnalytics(component, analyticsObj);
+        this.props.modules?.analytics.sendAnalytics(component, analyticsObj, uiElementProps);
```

Comment:
What about always propagating the UIElement props to `submitAnalytics`? 
Then we do not need to override this method in the AmazonPay/ApplePay etc..

In this case, we would:

```js
this.props.modules?.analytics.sendAnalytics(component, analyticsObj, this.props);
```

And then in the analytics we would use the prop if it is available or not. We would not need to  care about adding this method overload in the express PM's

Reply:
At the moment, instead of just throwing all of a component's props at the analytics call, all the time - I want a way to identify those components that _need_ to pass some of their props to the analytics endpoint, since I think this will be the exception rather than the rule.
If this looks to change in the future, if more components need to pass elements from their props, then I'll revisit this decision